Funcky/Extensions/StringExtensions.cs

```diff
@@ -0,0 +1,91 @@
+
+        public static Option<int> IndexOfAnyOrNone(this string haystack, char[] anyOf)
+            => MapIndexToOption(haystack.IndexOfAny(anyOf));
+
+        public static Option<int> IndexOfAnyOrNone(this string haystack, char[] anyOf, int startIndex)
+            => MapIndexToOption(haystack.IndexOfAny(anyOf, startIndex));
+
+        public static Option<int> IndexOfAnyOrNone(this string haystack, char[] anyOf, int startIndex, int count)
+            => MapIndexToOption(haystack.IndexOfAny(anyOf, startIndex, count));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, char value)
+            => MapIndexToOption(haystack.LastIndexOf(value));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, char value, int startIndex)
+            => MapIndexToOption(haystack.LastIndexOf(value, startIndex));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, char value, int startIndex, int count)
+            => MapIndexToOption(haystack.LastIndexOf(value, startIndex, count));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, string value)
+            => MapIndexToOption(haystack.LastIndexOf(value));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, string value, int startIndex)
+            => MapIndexToOption(haystack.LastIndexOf(value, startIndex));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, string value, int startIndex, int count)
+            => MapIndexToOption(haystack.LastIndexOf(value, startIndex, count));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, string value, StringComparison comparisonType)
+            => MapIndexToOption(haystack.LastIndexOf(value, comparisonType));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, string value, int startIndex, StringComparison comparisonType)
+            => MapIndexToOption(haystack.LastIndexOf(value, startIndex, comparisonType));
+
+        public static Option<int> LastIndexOfOrNone(this string haystack, string value, int startIndex, int count, StringComparison comparisonType)
+            => MapIndexToOption(haystack.LastIndexOf(value, startIndex, count, comparisonType));
+
+        public static Option<int> LastIndexOfAnyOrNone(this string haystack, char[] anyOf)
+            => MapIndexToOption(haystack.LastIndexOfAny(anyOf));
+
+        public static Option<int> LastIndexOfAnyOrNone(this string haystack, char[] anyOf, int startIndex)
+            => MapIndexToOption(haystack.LastIndexOfAny(anyOf, startIndex));
+
+        public static Option<int> LastIndexOfAnyOrNone(this string haystack, char[] anyOf, int startIndex, int count)
+            => MapIndexToOption(haystack.LastIndexOfAny(anyOf, startIndex, count));
+
+        private static Option<int> MapIndexToOption(int index)
+        {
+            const int sentinelValue = -1;
```

Comment:
maybe call it "NotFoundValue" and extract it as a static class constant.

Reply:
b0ec9d190f1eddb5271e82feb9a709df729cae4a